news/_posts/2019-12-25-prestashop-in-2019-and-beyond-part-3-the-future-architecture.md

```diff
@@ -0,0 +1,269 @@
+
+**If you want to get somewhere, _anywhere_, the first thing you have to do is decide where you want to go.** It's only _after that_ that you can start thinking on how you'll get there. 
+
+So we started working on that.
+
+When I first started talking about a new architecture during my first presentation before agencies at PrestaShop's Paris HQ back in 2017, attendants were rolling their eyes. "_Seriously?_", they said, "Are you seriously changing everything again? We're already struggling with 1.7 and now we'll have to spend money rebuilding our themes and modules _again_?". 
+
+I already knew by then that an architecture change isn't done overnight—especially for a development platform. So I reassured them by saying that it was our long-term vision and that they didn't have to worry about that right now, and that was the end of it.
+
+But even today, [four years after announcing 1.7 would be Symfony-based][introducing-symfony], many people are convinced that the Symfony migration was a waste of time and money, and that PrestaShop just did it to "force people into buying new modules". Some think that Symfony is a solution looking a problem to fix, a problem that didn't exist in the first place. I hope that after having read the previous two parts, you already understand that we're not doing it either, as someone once put it, "just because we enjoy beautiful code" (even if we, as most developers, actually _do_ enjoy beautiful code). There are many issues to fix,   yes, I know. But migrating to Symfony is a way of progressively giving more sense to PrestaShop's code, using modern tools, and improve its quality through incremental refactoring.
+
+Similarly, since last year we have been making architecture choices that may have seemed arbitrary or incomprehensible to some. "Why are you making it so needlessly complicated?", they said. Rest assured, **there is a good reason**. Each choice we make answers to a specific problem in the context of a bigger long-term vision. We just needed time to put this reasoning into words (or should I say, articles), so that everyone understands not just _what_ we are planning to do, but more importantly, _why_ we are leading the project in this direction.
+
+## What drives a new architecture?
+
+When we began thinking about a new architecture, we started with a simple exercise: if we had to build PrestaShop again, what would its architecture look like?
+
+### A solution for every problem
+
+Our first job was trying to understand what our current problems were and where they were coming from–this is what the previous articles were about. Once we knew what was dragging us back, we were able to start looking for answers.
+
+A new architecture should provide an answer to each one of PrestaShop's main structural problems.
+
+#### Interaction delays
```

Comment:
In the previous article you don't use these words but I find reference to this under "An aging architecture"

Reply:
Yeah, I was wondering whether I should use a similar wording or if it was too generic for what I was about to talk about. I think this one is more specific for this particular issue.